internal/deployment_client.go

```diff
@@ -0,0 +1,244 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package internal
+
+import (
+	"context"
+	"time"
+
+	commonpb "go.temporal.io/api/common/v1"
+)
+
+// DeploymentReachability specifies which category of tasks may reach a worker
+// associated with a deployment, simplifying safe decommission.
+// NOTE: Experimental
+type DeploymentReachability int
+
+const (
+	// DeploymentReachabilityUnspecified - Reachability level not specified.
+	// NOTE: Experimental
+	DeploymentReachabilityUnspecified = iota
+
+	// DeploymentReachabilityReachable - The deployment is reachable by new
+	// and/or open workflows. The deployment cannot be decommissioned safely.
+	// NOTE: Experimental
+	DeploymentReachabilityReachable
+
+	// DeploymentReachabilityClosedWorkflows - The deployment is not reachable
+	// by new or open workflows, but might be still needed by
+	// Queries sent to closed workflows. The deployment can be decommissioned
+	// safely if user does not query closed workflows.
+	// NOTE: Experimental
+	DeploymentReachabilityClosedWorkflows
+
+	// DeploymentReachabilityUnreachable - The deployment is not reachable by
+	// any workflow because all the workflows who needed this
+	// deployment are out of the retention period. The deployment can be
+	// decommissioned safely.
+	// NOTE: Experimental
+	DeploymentReachabilityUnreachable
+)
+
+type (
+	// Deployment identifies a set of workers. This identifier combines
+	// the deployment series name with their Build ID.
+	// NOTE: Experimental
+	Deployment struct {
+		// SeriesName - Name of the deployment series. Different versions of the same worker
+		// service/application are linked together by sharing a series name.
+		SeriesName string
+
+		// BuildID - identifies the worker's code and configuration version.
+		BuildID string
+	}
+
+	// DeploymentTaskQueueInfo describes properties of the Task Queues involved
+	// in a deployment.
+	// NOTE: Experimental
+	DeploymentTaskQueueInfo struct {
+		// Name - Task queue name.
+		Name string
+
+		// Type - The type of this task queue.
+		Type TaskQueueType
+
+		// FirstPollerTime - Time when the server saw the first poller for this task queue
+		// in this deployment.
+		FirstPollerTime time.Time
+	}
+
+	// DeploymentInfo holds information associated with
+	// workers in this deployment.
+	// Workers can poll multiple task queues in a single deployment,
+	// which are listed in this message.
+	// NOTE: Experimental
+	DeploymentInfo struct {
+		// Deployment - An identifier for this deployment.
+		Deployment Deployment
+
+		// CreateTime - When this deployment was created.
+		CreateTime time.Time
+
+		// IsCurrent - Whether this deployment is the current one for its deployment series.
+		IsCurrent bool
```

Comment:
```suggestion
		Current bool
```

Pedantic though, don't need to change

Reply:
It was `Current`, @ShahabT asked to change it to `IsCurrent`, it is a bool so it should be clear, but I don't have strong opinions...

Reply:
It is a bit strange to use `Is` prefixes in Go fields for these kinds of values IMO (unlike Java where of course you will)